components/form-control/script.js

```diff
@@ -169,24 +155,38 @@ class GamefaceFormControl extends HTMLElement {
      */
     serializeElementData(element, params) {
         if (element.hasAttribute('disabled')) return;
-        const tagName = element.tagName.toLowerCase();
+        return this.serializeSimpleElementData(element, params);
+    }
 
-        switch (tagName) {
-            case tags.INPUT:
-            case tags.TEXTAREA:
-                return this.serializeSimpleElementData(element, params);
-            case tags.GAMEFACE_SWITCH:
-                return this.serializeSwitchData(element, params);
-            case tags.GAMEFACE_RANGESLIDER:
-                return this.serializeRangeSliderData(element, params);
-            case tags.GAMEFACE_CHECKBOX:
-                return this.serializeCheckboxData(element, params);
-            case tags.GAMEFACE_DROPDOWN:
-                return this.serializeDropDownData(element, params);
-            case tags.GAMEFACE_RADIO_GROUP:
-                return this.serializeRadioGroupData(element, params);
-            default: break;
-        }
+    /**
+     * Creates a <gameface-tooltip> element, sets its message and shows it on top
+     * of given element.
+     *
+     * @param {string} message
+     * @param {HTMLElement} element
+    */
+    showError(error, element) {
+        const elementName = element.getAttribute('name');
+        const elementTagName = element.tagName.toLowerCase();
+        const id = `${elementTagName}-${elementName}-error-tooltip`;
```

Comment:
This is more complicated than it looks.
1. The example above conceptually changes the tooltip. The design of the tooltip was inspired by the html [label element where you let the label know which is its target element, not vice versa](https://www.w3schools.com/tags/tag_label.asp). - if we pass tooltip-id to the form element, we connect the form-element to the tooltip and it should be the opposite.
2. Setting a unique id is complicated because we change the element runtime and we have too manage unique ids. We 'll need to :
- keep track of the ids that are in use
- update the list if an element is removed
- make sure the ids don't collide with user defined ids
changing the target attribute requires change of 1.code, 2.tests and 3.documentation.

Using the name as identifies (as it is now) is wrong, I didn't consider that you could have the same names in a form.
But I think that the easiest solution is to pass the target element directly as a property of the tooltip, as you would normally do in a JS component : 

```
class Tooltip() {
    constructor(target) {
        target.addEventListenrer('click', this.show());
    }
}
...
new Tooltip(element)
```
because we can'tinstantiate HTMLElements we'll pass it as property.

-----------------------------------------------------------------------------------------------------------
Changed to:

in tooltip:
`this.triggerElement = this.targetElement || document.querySelector(this.elementSelector);`

in form:
```
this.tooltip = document.createElement('gameface-tooltip');
this.tooltip.targetElement = element;
```

Keeping the tooltip as property of the form is not ideal but it lets us keep the tooltip somewhat persistent. Otherwise we would have to find a way to identify tooltips, which is more complicated and error prone.

Reply:
That is a very good solution of the problem! I really like it! For now it is the best IMO.